src/actions/index.jsx

```diff
@@ -1,12 +1,43 @@
 import { auth, provider, storage } from "../firebase";
 import db from "../firebase";
+
 import {
   SET_DARK_MODE,
   SET_USER,
   SET_LOADING_STATUS,
   GET_ARTICLES,
 } from "./actionType";
 
+export function generateKeyWords(name, email) {
+  const words = [];
+
+  name.split(" ").forEach((word) => words.push(word));
+  words.push(email.split("@")[0]);
+
+  return words;
+}
+
+export async function setFbUser(userData) {
+  const userQuery = db.collection("users").doc(userData.email);
+
+  const data = {
+    connections: [],
+    keywords: generateKeyWords(userData.displayName, userData.email),
+    name: userData.displayName,
+    email: userData.email,
+    photoURL: userData.photoURL,
+    received_requests: [],
```

Comment:
Are these received and sent request for pending networking request? I have to think about this because I have a feeling we are missing something we need. I have not been given back my school supplies with my hand drawn wire frames and notes  yet. I will meditate on it.

Reply:
Yup, i tested it,  but im not 100% sure yet

Reply:
If the code dose not stop the stop the app from rendering and you just need to test it. Will you please write yourself some TODO: comments near it and push it, pretty please?